src/spark_rapids_ml/core.py

```diff
@@ -470,9 +449,41 @@ def _train_udf(pdf_iter: Iterator[pd.DataFrame]) -> pd.DataFrame:
             .mapPartitions(lambda x: x)
         )
 
-        if return_model == False:
-            return pipelined_rdd
+        return pipelined_rdd
+
+
+class _CumlEstimator(Estimator, _CumlCaller):
+    """
+    The common estimator to handle the fit callback (_fit). It should:
+    1. set the default parameters
+    2. validate the parameters
+    3. prepare the dataset
+    4. train and return CUML model
+    5. create the pyspark model
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    @abstractmethod
+    def _out_schema(self) -> Union[StructType, str]:
+        pass
+
+    @abstractmethod
+    def _create_pyspark_model(self, result: Row) -> "_CumlModel":
+        """
+        Create the model according to the collected Row
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def _get_cuml_fit_func(
```

Comment:
This is inherited from _CumlCaller now so don't think it is needed here again.

Reply:
Yeah. Confirmed it is not needed. Already have it deleted.